imap_processing/hi/l2/hi_l2.py

```diff
@@ -0,0 +1,54 @@
+"""IMAP-HI l2 processing module."""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.utils import load_cdf
+from imap_processing.ena_maps.ena_maps import PointingSet
+from imap_processing.hi.l1c.hi_l1c import N_SPIN_BINS
+from imap_processing.spice.geometry import SpiceFrame
+
+
+class HiPointingSet(PointingSet):
+    """
+    PointingSet object specific to Hi L1C PSet data.
+
+    Parameters
+    ----------
+    dataset : xarray.Dataset
+        Hi L1C pointing set data loaded in an xarray.DataArray.
+    """
+
+    def __init__(self, dataset: xr.Dataset):
+        self.spice_reference_frame = SpiceFrame.ECLIPJ2000
+        self.data = dataset
+        self.epoch = self.data["epoch"].values[0]
+        self.num_points = N_SPIN_BINS
+        self.az_el_points = np.column_stack(
+            (
+                np.squeeze(self.data["hae_longitude"]),
+                np.squeeze(self.data["hae_latitude"]),
+            )
+        )
+        self.spatial_coords = "spin_angle_bin"
+
+    @classmethod
+    def from_cdf(cls, cdf_path: str | Path) -> HiPointingSet:
```

Comment:
Would this be beneficial to put on the `PointingSet` superclass for other instruments to use too?

Reply:
This could potentially combine/replace `UltraPointingSet.from_path_or_dataset` if the reading-in is similar enough between the instruments.